app/controllers/basic_auth_controller.rb

```diff
@@ -6,7 +6,7 @@ class BasicAuthController < ApplicationController
   def create
     if user_signed_in?
       permission_checks = [params[:permission_name], "#{params[:permission_name]}.#{params[:format]}"]
-      groups = current_user.groups
+      groups = current_user.asserted_groups
```

Comment:
what's the difference between these two?

Reply:
Groups is the base relation between users and groups (when an admin adds a user to a group, they'd show up here), but asserted_groups is a [new function in this change](https://github.com/CentauriSolutions/EyeDP/pull/144/files#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29bR135-R142) that checks for 2fa enabled, and then potentially returns a subset of groups the user is a member of.